Warn if config changes without generation changing

hmusum opened the pull request.

Commit:
Unexpected config change: Lower log level and add explanation

hakonhall commented:
AFAIK we don't know how the config can change (.equals()) if the hashes are equal?

Commit:
Warn if config updated without generation updated

Always add response to response queue for subscription. Add warnings
if config changes without generation and of config state has an
unexpected change.
